cabot/cabotapp/views.py

```diff
@@ -597,7 +597,26 @@ class ServiceListView(LoginRequiredMixin, ListView):
     context_object_name = 'services'
 
     def get_queryset(self):
-        return Service.objects.all().order_by('name').prefetch_related('status_checks')
+        # we preload the check counts manually here to avoid an extra 4 DB hits per service in the template
+        # we use manual SQL because (as far as I can tell) the tools needed to express this in ORM language
+        # don't exist in django 1.6 :(
+        return Service.objects.all().order_by('name')\
+            .extra(select={
+                'active_checks_count':
+                    "SELECT COUNT(*)\n"
+                    "FROM cabotapp_statuscheck\n"
+                    "INNER JOIN cabotapp_service_status_checks "
```

Comment:
Thinking about it more...

```
Services.objects.all()\
  .annotate(active_checks_count=Count(status_checks__active=True))\
  .annotate(inactive_checks_count=Count(status_checks__active=False))
```

I would think this would work 🤔I'll give it a try later.

Reply:
I looked through the docs and I'm still pretty sure the only way to do this is with raw SQL.

I can clean up the references to the table names like this:
```
                'active_checks_count':
                    "SELECT COUNT(*)\n"
                    "FROM {statuscheck}\n"
                    "INNER JOIN {service_status_checks} "
                    "ON ({statuscheck}.id = {service_status_checks}.statuscheck_id)\n"
                    "WHERE {statuscheck}.active = TRUE\n"
                    "  AND service_status_checks.service_id = {service}.id".format(
                        statuscheck=StatusCheck._meta.db_table,
                        service_status_checks=Service.status_checks.through._meta.db_table,
                        service=Service._meta.db_table,
                    ),
```

I can also do the field names, but that basically amounts to `fieldname=model.fieldname._meta.get_field_by_name(fieldname).name` (which equals `fieldname`) which is super redundant and not really any more readable, IMO. How about some asserts instead, like `assert model.field is not None`?

Reply:
im good with the code snippet you have. Let's not do asserts (don't think asserts should be used for flow control) but we could throw exceptions or just write a test for the query?